discussions/secure_web_api.md

```diff
@@ -0,0 +1,119 @@
+
+## Types of certificates are supported
+
+The TLS protocol aims primarily to provide cryptography, including privacy (confidentiality), integrity, and
+authenticity through certificates, between L3AFD and client's communication. These certificates can be issued by a
+third-party trusted authority (i.e., IdenTrust, DigiCert, Sectigo, etc), and we can create self-signed certificates
+using tools like ```openssl```.
+
+This completely depends on the users, whether to use Trusted CA certificates or Self signed certificates. L3AF will not 
+provide any certificates.
+
+## L3AF deployment scenarios
+
+L3AF could be running in two scenarios, users can use L3AF in secure enterprise private networks and in public network.
+In case of private network, L3AFD and clients will be communicating with each other over a network that is normally
+protected by vpn or PCI (Payment Card Information), and hence it may not be essential to enable mTLS in this case.
+
+However, in case of public network, clients will be communicating with L3AFD over insecure networks. L3AF wants to
+secure its endpoints using industry's best standard available solutions. This can be configured from l3afd’s config file,
+and by default, mTLS will be disabled.
+
+## Enabling mTLS for L3AFD
+
+L3AFD (L3AF Daemon) will require a set of root certificates (root.crt and root.key) and a pair of server certificates
+(server.crt and server.key) to run in mTLS enabled mode. L3AFD will check for these certificates, and in case these are
+not found, L3AFD will stop with error ```certificates are not found```.
+
+The client will require a pair of client certificates (client.crt and client.key), generated from the same root
+certificates from which the server certificates have been generated, and the public root certificate (root.crt) to
+communicate with L3AFD.
+
+Process to enable mTLS 
+- Provision of Certificates 
+- Location of Certificates 
+- Enable mTLS
+
+### Provision of Certificates
+
+As mentioned before, users can use already pre-existing certificates. L3AF will not provide any certificates.
+
+### Location of Certificates
+
+The default location for the certificates can be ‘/etc/ssl/l3af/certs/’ and this path can also be changed through a
+l3afd configuration option in cases where the user would like to use a custom location.
+
+The user will have to place the root certificates and the server certificates in the configured directory path before
+starting L3AFD.
+
+### Enabling mTLS
+
+L3AFD will provide a flag to enable mTLS and this can be configured in l3afd.cfg, by default mTLS will be disabled
+in case l3afd is listening on localhost.
+
+## Minimum TLS version
+
+TLS version popularly supported in the market v1.2 and v1.3. L3AF can support v1.3 by default and require configuration
+to allow downgrading security.
+
+## L3AFD Web API Listening Interface
+
+L3AFD can be configured to listen on a different IP address / interface than localhost.
```

Comment:
Phrasing is bad since as worded it implies localhost is "a" IP address.  It's not.  Localhost is two addresses (one IPv4 and one IPv6).   As worded the "a" also implies that L3AFD can only listen on a single IP address, which would also be bad as that would prevent simultaneous uses of both IPv4 and IPv6.   Line 92 also implies that it can only listen on a single interface, which also seems too limiting to me.   Maybe these are statements of current limitations, not sure, but these are not desirable.